enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+    // +required
+    Spec UserDefinedNetworkSpec `json:"spec"`
+    // +optional
+    Status UserDefinedNetworkStatus `json:"status,omitempty"`
+}
+```
+Suggested API validation rules:
+- Spec defined as immutable 
+  <br>
+  Avoid incomplete state in a scenario where a NAD is created according to UDN spec, and pods connected to the network, 
+  UDN spec changes, now pods connected to a network that was created from previous revision spec.
+- `Subnets` are mandatory for `Layer3` topology.
+- `Localnet` topology is not supported for primary network.
+- `IPAMLifecycle` is supported for `Layer2` and `Localnet` topology.
+
+Suggested CRD short-name: `udn`
+
+```golang
+// UserDefinedNetworkSpec defines the desired state of UserDefinedNetwork.
+type UserDefinedNetworkSpec struct {
+    // The topological configuration for the network.
+    // +kubebuilder:validation:Enum=Layer2;Layer3;Localnet
+    Topology NetworkTopology `json:"topology"`
+    
+    // The network role in the pod (e.g.: Primary, Secondary).
+    // +kubebuilder:validation:Enum=Primary;Secondary
+    Role NetworkRole `json:"role"`
+    
+    // The maximum transmission unit (MTU).
+    // MTU is optional, if not provided the globally configured value in OVN-Kubernetes (defaults to 1400) is used for the network.    
+    // +optional
+    MTU uint `json:"mtu,omitempty"`
+
+
+    // The subnet to use for the pod network across the cluster.
+    //
+    // Dualstack clusters may set 2 subnets (one for each ip family), 
+    // otherwise only 1 subnet is allowed.
+    //
+    // When topology is `Layer3`, the given subnet is split into smaller subnets for every node.
+    // To specify how the subnet should be split, the following format is supported for `Layer3` network:
+    // `10.128.0.0/16/24`, which means that every host will get a `/24` subnet.
+    // If host subnet mask is not set (for example, `10.128.0.0/16`), it will be assigned automatically.
+    // 
+    // For `Layer2` and `Localnet` topology types, the format should match standard CIDR notation, without
+    // providing any host subnet mask.
+    // This field may be omitted for `Layer2` and `Localnet` topologies. 
+    // In that case the logical switch implementing the network only provides layer 2 communication,
+    // and users must configure IP addresses for the pods.
+    // Port security only prevents MAC spoofing
+    // +optional
+    Subnets []string `json:"subnets,omitempty"`
+    
+    // A list of CIDRs.
+    // IP addresses are removed from the assignable IP address pool and are never passed to the pods.
+    // +optional
+    ExcludeSubnets []string `json:"excludeSubnets,omitempty"`
+    
+    // Subnet used inside the OVN network topology.
+    // This field is ignored for non-primary networks (e.g.: Role Secondary).
```

Comment:
You can use validation to prevent it from being set, or create a discriminated union of primary/secondary